storage-manager/src/Replicator.cpp

```diff
@@ -121,9 +124,17 @@ void Replicator::printKPIs() const
 int Replicator::newObject(const boost::filesystem::path& filename, const uint8_t* data, off_t offset,
                           size_t length)
 {
-  int fd, err;
-  string objectFilename = msCachePath + "/" + filename.string();
+  const string objectFilename = msCachePath + "/" + filename.string();
+  auto kvStorage = KVStorageInitializer::getStorageInstance();
+  auto keyGen = std::make_shared<FDBCS::BoostUIDKeyGenerator>();
+  FDBCS::BlobHandler blobWriter(keyGen);
+  const std::string dataValue((char*)data);
+  if (!blobWriter.writeBlob(kvStorage, objectFilename, dataValue))
+  {
+    return -1;
+  }
 
+  int fd, err;
   OPEN(objectFilename.c_str(), O_WRONLY | O_CREAT);
```

Comment:
Do you need those file-based operations that take place after this line?

Reply:
Yes, this file operations for the cache files, of files loaded from S3